dandiapi/api/views/serializers.py

```diff
@@ -254,9 +254,24 @@ class AssetFolderSerializer(serializers.Serializer):
     modified = serializers.DateTimeField()
 
 
+class AssetFileSerializer(AssetSerializer):
+    class Meta(AssetSerializer.Meta):
+        fields = AssetSerializer.Meta.fields + ['url']
+
+    url = serializers.SerializerMethodField(method_name='get_url')
+
+    def get_url(self, asset: Asset):
+        if asset.is_blob:
+            return asset.blob.s3_url
+        elif asset.is_embargoed_blob:
+            return asset.embargoed_blob.s3_url
+        else:
+            return asset.zarr.s3_url
```

Comment:
IMO this abstraction should probably be done on the Asset model directly

```python
    @property
    def s3_url(self):
        if self.is_blob:
            return self.blob.s3_url
        elif self.is_embargoed_blob:
            return self.embargoed_blob.s3_url
        else:
            return self.zarr.s3_url
```

Then here, we could just do
```suggestion
    url = serializers.URLField(source='s3_url')
```